blockchains/xrp/xrp_worker.js

```diff
@@ -0,0 +1,175 @@
+          command: 'tx',
+          transaction: tx,
+          minLedgerVersion: ledger_index,
+          maxLedgerVersion: ledger_index
+        }).catch((error) => {
+          if (error.message === 'txnNotFound') {
+            return Promise.resolve(null);
+          }
+  
+          return Promise.reject(error);
+        })
+      );
+  
+      transactions = await Promise.all(transactions);
+  
+      transactions = transactions.filter(t => t);
+
+      return { ledger: ledger, transactions };
+    }
+
+    const result = await this.connectionSend(connection, {
+      command: 'ledger',
+      ledger_index: parseInt(ledger_index),
+      transactions: true,
+      expand: true
+    }).then(value => value.result.ledger);
+  
+    assert(result.closed === true);
+  
+    return { ledger: ledger, transactions: result.transactions };
+  }
+
+  checkAllTransactionsValid(ledgers) {
+    for (let indexLedger = 0; indexLedger < ledgers.length; indexLedger++) {
+      const transactions = ledgers[indexLedger].transactions;
+      const blockNumber = ledgers[indexLedger].ledger.ledger_index;
+      for (let index = 0; index < transactions.length; index++) {
+        const transaction = transactions[index];
+        if('validated' in transaction && !transaction.validated) {
+          logger.error(`Transaction ${transaction.hash} at index ${index} in block ${blockNumber} is not validated. Aborting.`);
+          process.exit(-1);
+        }
+        if(!('meta' in transaction) && !('metaData' in transaction)) {
+          logger.error(`Transaction ${transaction.hash} at index ${index} in block ${blockNumber} is missing 'meta' field. Aborting.`);
+          process.exit(-1);
+        }
+      }
+    }
+  }
+
+  async work() {
+    if (this.lastConfirmedBlock === this.lastExportedBlock) {
+      this.sleepTimeMsec = constants.LOOP_INTERVAL_CURRENT_MODE_SEC * 1000;
+      const lastValidatedLedger = await this.connectionSend(this.connections[0], {
+        command: 'ledger',
+        ledger_index: 'validated',
+        transactions: true,
+        expand: false
+      });
+      const newConfirmedBlock = parseInt(lastValidatedLedger.result.ledger.ledger_index) - constants.CONFIRMATIONS;
+      if (newConfirmedBlock === this.lastConfirmedBlock) {
+        return [];
+      }
+      this.lastConfirmedBlock = newConfirmedBlock;
+    } else {
+      this.sleepTimeMsec = 0;
+    }
+    let toBlock = Math.min(this.lastExportedBlock + constants.SEND_BATCH_SIZE, this.lastConfirmedBlock);
+    let fromBlock = this.lastExportedBlock + 1;
+
+    const requests = [];
+    let transfers = [];
+    logger.info(`Fetching transfers for interval ${fromBlock}:${toBlock}`);
+    while (fromBlock + requests.length <= toBlock) {
+      const ledgerToDownload = fromBlock + requests.length;
+      requests.push(
+        await this.fetchLedgerTransactions(this.connections[ledgerToDownload % this.connections.length], ledgerToDownload)
```

Comment:
You shouldn't `await` here as this would cause requests to be evaluated sequentially, the goal here is to create parallelism.